snappiershot/snapshot/_raises.py

```diff
@@ -0,0 +1,78 @@
+""" Context class used to track and snapshot raised exceptions.
+
+Mimics and is inspired by `pytest.raises`.
+"""
+from types import TracebackType
+from typing import (
+    TYPE_CHECKING,
+    Generic,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    # Used to avoid circular imports.
+    from .snapshot import Snapshot
+
+_E = TypeVar("_E", bound=BaseException)
```

Comment:
could use a comment here explaining what these are / why needed (esp for other developers less familiar w/ type hinting)